cmd/jaeger/internal/exporters/storageexporter/sanitizer/emptyservicename.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package sanitizer
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/pkg/otelsemconv"
+)
+
+const (
+	emptyServiceName   = "empty-service-name"
+	missingServiceName = "missing-service-name"
+)
+
+// NewEmptyServiceNameSanitizer returns a sanitizer function that replaces
+// empty and missing service names with placeholder strings.
+func NewEmptyServiceNameSanitizer() Sanitize {
+	return sanitizeEmptyServiceName
+}
+
+func sanitizeEmptyServiceName(traces ptrace.Traces) ptrace.Traces {
+	resourceSpans := traces.ResourceSpans()
+	for i := 0; i < resourceSpans.Len(); i++ {
+		resourceSpan := resourceSpans.At(i)
+		attributes := resourceSpan.Resource().Attributes()
+		serviceName, ok := attributes.Get(string(otelsemconv.ServiceNameKey))
+		fmt.Printf("service name: %s, ok: %v\n", serviceName.Str(), ok)
+		if !ok {
+			attributes.PutStr(string(otelsemconv.ServiceNameKey), missingServiceName)
+		} else if serviceName.Str() == "" {
```

Comment:
what if it's not a string attribute?

Reply:
should we change this check to be the following?
```go
if !ok {
	attributes.PutStr(string(otelsemconv.ServiceNameKey), missingServiceName)
} else if serviceName.Type() != pcommon.ValueTypeStr || serviceName.Str() == "" {
	attributes.PutStr(string(otelsemconv.ServiceNameKey), emptyServiceName)
}
```

Reply:
If it's a different type it's a different error message